### PR TITLE
fix: switch to standard decorators and tighten property types

### PR DESCRIPTION
**Pull request by kyubisation**
This PR switches from experimental decorators to standard decorators. It additionally tightens the types of properties and applies `forceType` where necessary.

### CODE REVIEW

File: src/visual-regression-app/src/components/test-case/image-diff/image-diff.ts [hunk 1]
```diff
@@ -23,16 +24,19 @@ const getImageDimension = (img: HTMLImageElement): string =>
...
-export class ImageDiff extends LitElement {
+class ImageDiff extends LitElement {
   public static override styles: CSSResultGroup = style;
 
-  @property() public screenshotFiles?: ScreenshotFiles;
+  @property() public accessor screenshotFiles: ScreenshotFiles | null = null;
 
-  @state() private _baselineDimension?: string;
-  @state() private _failedDimension?: string;
+  @state() private accessor _baselineDimension!: string;
```
Comment (jeripeierSBB) [inline src/visual-regression-app/src/components/test-case/image-diff/image-diff.ts:34]: same topic, could we still go with "?" for @ state ?

File: src/visual-regression-app/src/components/test-case/test-case.ts [hunk 2]
```diff
@@ -35,16 +35,17 @@ interface Filter {
...
   public static override styles: CSSResultGroup = style;
 
-  @property() public params?: { componentName: string; testCaseName: string };
+  @property() public accessor params: { componentName: string; testCaseName: string } | null = null;
 
-  @state() private _testCase?: ScreenshotTestCase;
-  @state() private _testCaseIndex: number = -1;
-  @state() private _filter: Filter = {};
-  @state() private _showGlobalDiff = true;
+  @state() private accessor _testCase!: ScreenshotTestCase;
```
Comment (jeripeierSBB) [inline src/visual-regression-app/src/components/test-case/test-case.ts:45]: and the last time the same question ;). would `?` still work?

**Review by jeripeierSBB — COMMENTED** [review]
Great work. Thanks a lot for your effort, especially for linting everything :). Looks good, I have some comments which are basically questions :)

File: src/elements/core/decorators/omit-empty-converter.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { defaultConverter, type ComplexAttributeConverter } from 'lit';
```
[the remark this answers is not in the document]
Comment (kyubisation) [reply src/elements/core/decorators/omit-empty-converter.ts:1]: Discussed as acceptable.

**Pull request merged**